pydantic/dataclasses.py

```diff
@@ -105,33 +108,28 @@ def _pydantic_post_init(self: 'DataclassType', *initvars: Any) -> None:
     return cls
 
 
-if TYPE_CHECKING:
-    # see https://github.com/python/mypy/issues/6239 for explanation of why we do this
-    from dataclasses import dataclass as dataclass
-else:
-
-    def dataclass(
-        _cls: Optional[AnyType] = None,
-        *,
-        init: bool = True,
-        repr: bool = True,
-        eq: bool = True,
-        order: bool = False,
-        unsafe_hash: bool = False,
-        frozen: bool = False,
-        config: Type['BaseConfig'] = None,
-    ) -> Union[Callable[[AnyType], 'DataclassType'], 'DataclassType']:
-        """
-        Like the python standard lib dataclasses but with type validation.
-
-        Arguments are the same as for standard dataclasses, except for validate_assignment which has the same meaning
-        as Config.validate_assignment.
-        """
-
-        def wrap(cls: AnyType) -> 'DataclassType':
-            return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen, config)
-
-        if _cls is None:
-            return wrap
-
-        return wrap(_cls)
+def dataclass(
+    _cls: Optional[AnyType] = None,
+    *,
+    init: bool = True,
+    repr: bool = True,
+    eq: bool = True,
+    order: bool = False,
+    unsafe_hash: bool = False,
+    frozen: bool = False,
+    config: Type['BaseConfig'] = None,
```

Comment:
We usually give a `Config` class as a `config` argument.
But, The class is not inherited from `BaseConfig`.
Should we change the type to  `Type`?

Reply:
Yes, I think it should just be `Type`

Reply:
Thank you for your reply.
Mypy rejects `Type`  which is a generic type that needs an argument.
I changed `Type['BaseConfig']` to `Type[Any]`.